### PR TITLE
Initial deployment of the app to Kubernetes

### PR DESCRIPTION
**Pull request by GhiocelAndrei**

### CODE REVIEW

File: deployment.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        app.kubernetes.io/name: issue-tracker-app
+    spec:
+      containers:
+      - name: issue-tracker-app-container
+        image: ghiocel/issue-tracker:v1
+        ports:
+        - containerPort: 80
+        env:
+        - name: ConnectionStrings__SqlServer
+          value: Data Source=host.docker.internal\SQLEXPRESS; Initial Catalog=IssueTrackerDb; User ID=sa;Password=1qazXSW@; trustServerCertificate=true
```
Comment (cristi-paraschiv) [inline deployment.yml:29]: this needs to be put in a secret

**Comment by cristi-paraschiv** [comment]
let's use a Helm chart and please put everything in a folder, such as `deploy`

File: deploy/values.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+appName: issue-tracker-app
+
+image:
+  name: ghiocel/issue-tracker
+  tag: v1
+
+port: 7141
```
Comment (cristi-paraschiv) [inline deploy/values.yaml:7]: let's get rid of this and use the default (80) everywhere

File: deploy/templates/service.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: "{{ .Values.appName }}-service"
+spec:
+  selector:
+    app: {{ .Values.appName }}
+  ports:
+    - protocol: TCP
+      port: {{ .Values.port }}
+      targetPort: 80
+  type: LoadBalancer
```
Comment (cristi-paraschiv) [inline deploy/templates/service.yaml:12]: let's use [NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) here;
we're not going to deploy this in the cloud (yet), so we don't have a LoadBalancer to put in front of it

**Review by cristi-paraschiv — COMMENTED** [review]
-

**Review by cristi-paraschiv — APPROVED** [review]

File: deployment.yml (hunk 1 above)
Comment (GhiocelAndrei) [reply deployment.yml]: Moved Connection String to secret

File: deploy/templates/service.yaml (hunk 3 above)
Comment (GhiocelAndrei) [reply deploy/templates/service.yaml]: Modified LoadBalancer to NodePort

**Pull request merged**